src/components/forms/VpnServerForm.js

```diff
@@ -0,0 +1,424 @@
+import React, { Component } from "react";
+import { NumberInput, TextArea } from "@carbon/react";
+import { titleCase, transpose, isNullOrEmptyString } from "lazy-z";
+import PropTypes from "prop-types";
+import { invalidCRNs } from "../../lib";
+import { isIpStringInvalid, isRangeInvalid } from "../../lib/iam-utils";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import { IcseSelect } from "../Dropdowns";
+import { IcseTextInput, IcseNameInput, IcseToggle } from "../Inputs";
+import { SecurityGroupMultiSelect } from "../MultiSelects";
+import { IcseFormGroup } from "../Utils";
+import IcseFormTemplate from "../IcseFormTemplate";
+import VpnServerRouteForm from "./VpnServerRouteForm";
+
+class VpnServerForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...this.props.data };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleNumberInputChange = this.handleNumberInputChange.bind(this);
+    this.handleMultiSelectChange = this.handleMultiSelectChange.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.handleAllowedIps = this.handleAllowedIps.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+  }
+
+  handleInputChange(event) {
+    let { name, value } = event.target;
+    let newState = { ...this.state };
+    //handle crn inputs
+    let crnList = value
+      ? value
+          .replace(/\s\s+/g, "") // replace extra spaces
+          .replace(/,(?=,)/g, "") // prevent null tags from
+          .replace(/[^\w,-:]/g, "")
+      : [];
```

Comment:
should these default to an array? both of the proptypes for certiciate_crn and client_ca_crn are string